pkg/operator/ceph/cluster/watcher.go

```diff
@@ -53,62 +55,65 @@ func (c *clientCluster) onK8sNode(object runtime.Object) bool {
-		logger.Debugf("node watcher: skipping cluster update. added node %q is unschedulable", node.Labels[v1.LabelHostname])
-		return false
-	}
-
-	if !k8sutil.NodeIsTolerable(*node, cephv1.GetOSDPlacement(cluster.Spec.Placement).Tolerations, false) {
-		logger.Debugf("node watcher: node %q is not tolerable for cluster %q, skipping", node.Name, cluster.Namespace)
-		return false
-	}
-
-	if !cluster.Spec.Storage.UseAllNodes {
-		logger.Debugf("node watcher: do not use all Nodes in cluster %q, skipping", cluster.Namespace)
-		return false
-	}
+		if !k8sutil.GetNodeSchedulable(*node) {
+			logger.Debugf("node watcher: skipping cluster update. added node %q is unschedulable", node.Labels[v1.LabelHostname])
+			return false
+		}
 
-	// Too strict? this replaces clusterInfo == nil
-	if cluster.Status.Phase != cephv1.ConditionReady {
-		logger.Debugf("node watcher: cluster %q is not ready. skipping orchestration", cluster.Namespace)
-		return false
-	}
+		if !k8sutil.NodeIsTolerable(*node, cephv1.GetOSDPlacement(cluster.Spec.Placement).Tolerations, false) {
+			logger.Debugf("node watcher: node %q is not tolerable for cluster %q, skipping", node.Name, cluster.Namespace)
+			return false
+		}
 
-	valid, _ := k8sutil.ValidNode(*node, cephv1.GetOSDPlacement(cluster.Spec.Placement))
-	if valid {
-		nodeName := node.Name
-		hostname, ok := node.Labels[v1.LabelHostname]
-		if ok && hostname != "" {
-			nodeName = hostname
+		if !cluster.Spec.Storage.UseAllNodes {
+			logger.Debugf("node watcher: do not use all Nodes in cluster %q, skipping", cluster.Namespace)
+			return false
 		}
 
-		// Make sure we can call Ceph properly
-		// Is the node in the CRUSH map already?
-		// If so we don't need to reconcile, this is done to avoid double reconcile on operator restart
-		// Assume the admin key since we are watching for node status to create OSDs
-		clusterInfo := cephclient.AdminClusterInfo(cluster.Namespace)
-		osds, err := cephclient.GetOSDOnHost(c.context, clusterInfo, nodeName)
-		if err != nil {
-			// If it fails, this might be due to the the operator just starting and catching an add event for that node
-			logger.Debugf("failed to get osds on node %q", nodeName)
+		// Too strict? this replaces clusterInfo == nil
+		if cluster.Status.Phase != cephv1.ConditionReady {
+			logger.Debugf("node watcher: cluster %q is not ready. skipping orchestration", cluster.Namespace)
 			return false
 		}
 
-		// If they are OSDs in the CRUSH map and if the host exists in the CRUSH map, don't reconcile
-		if osds != "" {
-			// This is Debug level because the node receives frequent updates and this will pollute the logs
-			logger.Debugf("node watcher: node %q is already an OSD node with %q", nodeName, osds)
-		} else {
-			logger.Infof("node watcher: adding node %q to cluster %q", node.Labels[v1.LabelHostname], cluster.Namespace)
-			return true
+		valid, _ := k8sutil.ValidNode(*node, cephv1.GetOSDPlacement(cluster.Spec.Placement))
+		if valid {
```

Comment:
nit: If you return immediately for the negative case, you can avoid the deeper nesting of the code block below
```suggestion
		if !valid {
			return false
		}
```